window.c

```diff
@@ -486,10 +488,21 @@ void win_draw_rect(win_t *win, int x, int y, int w, int h, bool fill, int lw,
 		XDrawRectangle(win->env.dpy, win->buf.pm, gc, x, y, w, h);
 }
 
-void win_set_title(win_t *win, const char *title)
+void win_set_title(win_t *win, const char *path)
 {
-	XStoreName(win->env.dpy, win->xwin, title);
-	XSetIconName(win->env.dpy, win->xwin, title);
+	char title[PATH_MAX];
+	const char *basename = strrchr(path, '/');
```

Comment:
A bit hacky, but I'm quite sure `/` is not allowed to be inside a filename so we should be fine.

One edge case I can think of is if path exceeds `PATH_MAX`. Not sure what happens then.

Reply:
> A bit hacky, but I'm quite sure `/` is not allowed to be inside a filename so we should be fine.

Yes.

> One edge case I can think of is if path exceeds `PATH_MAX`. Not sure what happens then.

I don't think it is our problem if it exceeds `PATH_MAX`.

Reply:
Anyways, don't think we should use PATH_MAX blindly. Should either malloc or use VLA.